### PR TITLE
Frontpage dialogue users

### PR DESCRIPTION
**Pull request by lagerros**
* Adds recommended users to dialogue with to the frontpage, together with associated styling
* Add a frontpage customisation icon where the user can choose which dialogue-related items to display or not (together with associated schemas)
* Does some refactoring of components into smaller components 

Assorted screenshots below!

image

image

NOTE: all of this topic data is synthetic. It should neither be implied that these people do nor that they do not hold these beliefs. 

image

image

image

image

### CODE REVIEW

File: packages/lesswrong/server/repos/UsersRepo.ts [hunk 1]
```diff
@@ -406,6 +405,36 @@ export default class UsersRepo extends AbstractRepo<DbUser> {
...
+          "userId" = uv._id
+          AND "parentCommentId" IN (
+            SELECT _id
+            FROM "Comments"
+            WHERE "userId" = $1
+          )
+      )
+    LIMIT $3 
+  `, [userId, upvotedUserIds, limit]);
+  }
```
Comment (b0b3rt) [inline packages/lesswrong/server/repos/UsersRepo.ts:436]: I think I remember this getting a comment from Ollie on the previous PR... would go back and check

File: packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+      marginLeft: 'auto',
+      minWidth: '25px',
+    },
+    color: theme.palette.text.dim3,
+    fontFamily: theme.palette.fonts.sansSerifStack,
+  },
+  dialogueRightContainer: {
+    display: 'flex',
+    justifyContent: 'space-between',
+    //paddingRight: 10,
```
Comment (rjmk) [inline packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:120]: Prolly wanna clean up commented-out code here and elsewhere

File: packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+  rowProps: DialogueUserRowProps<boolean>; 
+  classes: ClassesType<typeof styles>; 
+  showSuggestedTopics: boolean;
+}
+
+const DialogueRecommendationRow = ({ rowProps, classes, showSuggestedTopics }: DialogueRecommendationRowProps) => {
+  const { DialogueCheckBox, UsersName, PostsItem2MetaInfo, LWTooltip, ReactionIcon } = Components
+
+  const { targetUser, checkId, userIsChecked, userIsMatched } = rowProps;
+  const { captureEvent } = useTracking(); //it is virtuous to add analytics tracking to new components
```
Comment (rjmk) [inline packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:146]: I think it's only virtuous if you call `captureEvent`, which you don't appear to do

File: packages/lesswrong/components/dialogues/DialoguesList.tsx [hunk 4]
```diff
@@ -180,71 +276,60 @@ const DialoguesList = ({ classes }: { classes: ClassesType<typeof styles> }) =>
...
-                    currentUser={currentUser}
-                  />
-                </div>
-              </div>
-            </div>);
-          })}
-        </div>
+        {showReciprocityRecommendations && <div>
+          { (currentUser?.showMatches || currentUser?.showRecommendedPartners ) &&
+            <div className={classes.explanatoryNoteBox}>
```
Comment (rjmk) [inline packages/lesswrong/components/dialogues/DialoguesList.tsx:306]: This is nice, but kind of ends up looking like it applies to the whole Dialogue section?

Comment (rjmk) [reply packages/lesswrong/components/dialogues/DialoguesList.tsx:306]: image

File: packages/lesswrong/components/dialogues/DialoguesList.tsx [hunk 5]
```diff
@@ -106,26 +99,110 @@ const styles = (theme: ThemeType) => ({
...
+    color: theme.palette.primary.main,
+    cursor: 'pointer',
+    '&:hover': {
+      color: theme.palette.primary.light,
+    }
   }
 });
 
+interface DialogueMatchRowProps {
+  rowProps: DialogueUserRowProps<boolean>;
```
Comment (b0b3rt) [inline packages/lesswrong/components/dialogues/DialoguesList.tsx:151]: I would not nest props like this; it can be fine to have props which are themselves objects with fields but you should only do that if there's some reason to do that.  If you just want `DialogueMatchRowProps` to be a superset of `DialogueUserRowProps`, just say `interface DialogueMatchRowProps extends DialogueUserRowProps<boolean>`

File: packages/lesswrong/components/dialogues/DialoguesSectionFrontpageSettings.tsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+    [theme.breakpoints.down('xs')]: {
+      display: "flex",
+      flexDirection: "row",
+      alignItems: "center",
+    },
+  }
+})
+
+const DialoguesSectionFrontpageSettings = ({persistentSettings, hidden, currentShowDialogues, currentShowMyDialogues, currentShowMatches, currentShowRecommendedPartners, hideReciprocityButtons, classes}: {
+  persistentSettings?: any,
```
Comment (b0b3rt) [inline packages/lesswrong/components/dialogues/DialoguesSectionFrontpageSettings.tsx:47]: Should this be a boolean?

File: packages/lesswrong/components/dialogues/DialoguesSectionFrontpageSettings.tsx [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+  currentShowMatches: boolean,
+  currentShowRecommendedPartners: boolean,
+  hideReciprocityButtons: boolean
+  classes: ClassesType,
+}) => {
+  const { MetaInfo } = Components
+  const currentUser = useCurrentUser();
+  const updateCurrentUser = useUpdateCurrentUser();
+
+  const setSetting = (type: keyof typeof USER_SETTING_NAMES, newSetting: any) => {
```
Comment (b0b3rt) [inline packages/lesswrong/components/dialogues/DialoguesSectionFrontpageSettings.tsx:60]: `newSetting` should probably be a `boolean`?

File: packages/lesswrong/components/hooks/useDialogueMatchmaking.ts [hunk 8]
```diff
@@ -43,6 +46,15 @@ export const useDialogueMatchmaking = <T extends MatchmakingProps>(props: T): Us
...
   `, { skip: skipByDefault || !getMatchedUsers });
 
+  const recommendedUsersQueryResult = useQuery(gql`
+  query GetDialogueMatchedUsers {
+    GetDialogueRecommendedUsers {
+      _id
+      displayName
+    }
+  }
+  `, { skip: !getRecommendedUsers || !dialogueMatchmakingEnabled.get() });
```
Comment (b0b3rt) [inline packages/lesswrong/components/hooks/useDialogueMatchmaking.ts:56]: use `skipByDefault` rather than `dialogueMatchmakingEnabled.get()`, which also checks that there's a `currentUser` (i.e. this is a logged-in user).  I don't think we want to be making this query for logged out users?

File: packages/lesswrong/lib/collections/users/schema.ts [hunk 9]
```diff
@@ -1450,6 +1450,54 @@ const schema: SchemaType<DbUser> = {
...
+  showMyDialogues: {
+    type: Boolean,
+    canRead: [userOwns, 'sunshineRegiment', 'admins'],
+    canCreate: ['members'],
+    canUpdate: [userOwns, 'sunshineRegiment', 'admins'],
+    optional: false,
+    nullable: false,
+    group: formGroups.siteCustomizations,
+    hidden: !isLW,
+    label: "Show a list of recently active dialogues inside the frontpage widget",
```
Comment (b0b3rt) [inline packages/lesswrong/lib/collections/users/schema.ts:1474]: this label seems wrong

File: packages/lesswrong/lib/collections/users/schema.ts [hunk 10]
```diff
@@ -1450,6 +1450,54 @@ const schema: SchemaType<DbUser> = {
...
+  showRecommendedPartners: {
+    type: Boolean,
+    canRead: [userOwns, 'sunshineRegiment', 'admins'],
+    canCreate: ['members'],
+    canUpdate: [userOwns, 'sunshineRegiment', 'admins'],
+    optional: false,
+    nullable: false,
+    group: formGroups.siteCustomizations,
+    hidden: !isLW,
+    label: "Show a list of recently active dialogues inside the frontpage widget",
```
Comment (b0b3rt) [inline packages/lesswrong/lib/collections/users/schema.ts:1498]: this label seems wrong

File: packages/lesswrong/lib/collections/users/schema.ts [hunk 11]
```diff
@@ -1450,6 +1450,54 @@ const schema: SchemaType<DbUser> = {
     label: "Opted-in to receiving invitations for dialogue facilitation from LessWrong team",
```
Comment (b0b3rt) [inline packages/lesswrong/lib/collections/users/schema.ts:1450]: While you're here, can you put this into a form group so it's not hanging out at the top up here?  (Or just put it into a separate PR right now)

File: packages/lesswrong/lib/components.ts [hunk 12]
```diff
@@ -444,6 +444,7 @@ importComponent("DialogueEditorGuidelines", () => require('../components/posts/d
 importComponent("DialogueEditorFeedback", () => require('../components/posts/dialogues/DialogueEditorFeedback'));
 importComponent("NewDialogueDialog", () => require('../components/posts/NewDialogueDialog'));
 importComponent("DialogueSubmit", () => require('../components/posts/dialogues/DialogueSubmit'));
+importComponent("PostsListSettings", () => require('../components/posts/PostsListSettings'));
```
Comment (b0b3rt) [inline packages/lesswrong/lib/components.ts:447]: Was this previously not getting imported into the component map?

File: packages/lesswrong/server/repos/UsersRepo.ts [hunk 13]
```diff
@@ -406,6 +405,47 @@ export default class UsersRepo extends AbstractRepo<DbUser> {
...
+    return this.any(`
+      SELECT u.*
+      FROM unnest($2::text[]) AS uv(_id)
+      INNER JOIN "Users" AS u ON uv._id = u._id
+      WHERE
+        -- Exclude users that the current user has already checked
+        (
+          SELECT COUNT(*)
+          FROM "DialogueChecks"
+          WHERE "userId" = $1 AND "targetUserId" = uv._id AND "checked" = TRUE
```
Comment (b0b3rt) [inline packages/lesswrong/server/repos/UsersRepo.ts:420]: always prefer `IS TRUE` to `= TRUE`; regular boolean equality comparisons fall over on null values.

File: packages/lesswrong/server/resolvers/userResolvers.ts [hunk 14]
```diff
@@ -535,3 +535,24 @@ defineQuery({
...
+  name: "GetDialogueRecommendedUsers",
+  resultType: "[User]!",
+  fn: async (root, _, context) => {
+    const { currentUser } = context
+    if (!currentUser) {
+      throw new Error('User must be logged in to get recommended users');
+    }
+
+    const upvotedUsers = await context.repos.users.getUsersTopUpvotedUsers(currentUser, 50, 720)
+    let recommendedUsers = await context.repos.users.getDialogueRecommendedUsers(currentUser._id, upvotedUsers);
```
Comment (b0b3rt) [inline packages/lesswrong/server/resolvers/userResolvers.ts:549]: `recommendedUsers` isn't getting reassigned, so it doesn't need to be `let` rather than `const`

Comment (b0b3rt) [reply packages/lesswrong/server/resolvers/userResolvers.ts:549]: (well, I just saw it get reassigned down below... but you don't need to, you can just slice `shuffled`)

Comment (b0b3rt) [reply packages/lesswrong/server/resolvers/userResolvers.ts:549]: Actually I think you can just `ORDER BY random()` in a postgresql query so you probably don't need most of this logic at all.

File: packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx [hunk 15]
```diff
@@ -0,0 +1,230 @@
...
+                {topic.theirVote === 'agree' && [<ReactionIcon key={index} size={13} react={"agree"} />, <span key={index} className={ isExpanded ? classes.agreeText : classNames(classes.agreeText, classes.agreeTextCollapsedMobile) }>agrees: </span>, `"${topic.comment.contents.plaintextMainText}"`] } 
+                {topic.theirVote === 'disagree' && [<ReactionIcon key={index} size={13} react={"disagree"} />, <span key={index} className={ isExpanded ? classes.agreeText : classNames(classes.agreeText, classes.agreeTextCollapsedMobile) }>disagrees: </span>, `"${topic.comment.contents.plaintextMainText}"`] }
+              </p>
+            </div>
+          ))}
+          
+      </div>}
+      <div className={classes.dialogueRightContainer}>
+        {<span className={isExpanded ? classes.hideIcon : classes.expandIcon} onClick={toggleExpansion}>
+          {isExpanded ? '▲ hide' : (numHidden > 0 ? [`▼`, <span key={targetUser._id} className={classes.bigScreenExpandNote}> ...more</span>] : '')}
```
Comment (b0b3rt) [inline packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:212]: Nested ternaries are very difficult to read.  Please extract into an assignment to a variable before the `return` (which can just call a function with 2 if statements, or something)

File: packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx [hunk 16]
```diff
@@ -0,0 +1,230 @@
...
+            />
+          </PostsItem2MetaInfo>
+        </div>
+        {showSuggestedTopics && <div className={classes.topicRecommendationsList}>
+          {topicRecommendations?.slice(0,numShown).map((topic, index) => (
+            <div key={index} className={classes.suggestionRow}>
+              <p key={index} className={ isExpanded ? classes.debateTopicExpanded : classes.debateTopicCollapsed}>
+                {topic.theirVote === 'agree' && [<ReactionIcon key={index} size={13} react={"agree"} />, <span key={index} className={ isExpanded ? classes.agreeText : classNames(classes.agreeText, classes.agreeTextCollapsedMobile) }>agrees: </span>, `"${topic.comment.contents.plaintextMainText}"`] } 
+                {topic.theirVote === 'disagree' && [<ReactionIcon key={index} size={13} react={"disagree"} />, <span key={index} className={ isExpanded ? classes.agreeText : classNames(classes.agreeText, classes.agreeTextCollapsedMobile) }>disagrees: </span>, `"${topic.comment.contents.plaintextMainText}"`] }
+              </p>
```
Comment (b0b3rt) [inline packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:205]: This is not how we do conditional classNames; see any usage of the `classNames` function in e.g. DialogueMatchingPage.  (You do
```
classNames(classes.unconditionalClassName, {
  [classes.conditionalClassName]: booleanVariableWhichIsTrueIfClassShouldBeAppliedAndFalseOtherwise
})
```

File: packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx [hunk 17]
```diff
@@ -0,0 +1,230 @@
...
+  const preTopicRecommendations: {comment: {_id: string, contents: {html: string, plaintextMainText: string}}, recommendationReason: string, yourVote: string, theirVote: string}[] | undefined = topicData?.GetTwoUserTopicRecommendations; 
+  const topicRecommendations = preTopicRecommendations?.filter(topic => topic.theirVote !== null);
+  const numRecommendations = topicRecommendations?.length || 0;
+  const numShown = isExpanded ? numRecommendations : 1
+  const numHidden = Math.max(0, numRecommendations - numShown);
+
+  return (
+    <div>
+      <div key={targetUser._id} className={ isExpanded ? classNames(classes.dialogueUserRow, classes.dialogueUserRowExpandedMobile) : classes.dialogueUserRow}>
+        <div className={ (isExpanded || numRecommendations === 0) ? classNames(classes.dialogueLeftContainer, classes.dialogueLeftContainerExpandedMobile, classes.dialogueLeftContainerNoTopics) : classes.dialogueLeftContainer}>
```
Comment (b0b3rt) [inline packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:182]: see https://github.com/ForumMagnum/ForumMagnum/pull/8322/files#r1408663772

File: packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx [hunk 18]
```diff
@@ -0,0 +1,230 @@
...
+        yourVote
+        theirVote
+      }
+    }
+  `, {
+    variables: { userId: currentUser?._id, targetUserId: targetUser._id, limit:4 },
+  });
+
+  if (!currentUser) return <></>;
+  const preTopicRecommendations: {comment: {_id: string, contents: {html: string, plaintextMainText: string}}, recommendationReason: string, yourVote: string, theirVote: string}[] | undefined = topicData?.GetTwoUserTopicRecommendations;
```
Comment (b0b3rt) [inline packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:173]: please define an interface for this

Comment (b0b3rt) [reply packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:173]: Actually isn't this just `TopicRecommendation`?  If so you can move it from the server to the client (since you can't import from server files in client files)

File: packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx [hunk 19]
```diff
@@ -0,0 +1,230 @@
+// TODO: Import component in components.ts
```
Comment (b0b3rt) [inline packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:1]: clean up this TODO if it's done

File: packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx [hunk 20]
```diff
@@ -0,0 +1,230 @@
...
+import React, {useState} from 'react';
+import { registerComponent, Components } from '../../lib/vulcan-lib';
+import { useTracking } from "../../lib/analyticsEvents";
+import {useCurrentUser} from '../common/withUser';
+import {gql, useQuery} from '@apollo/client';
+import {DialogueUserRowProps} from '../users/DialogueMatchingPage';
+import classNames from 'classnames';
+
+const styles = (theme: ThemeType) => ({
+  dialogueUserRow: { // TODO; shared import
```
Comment (b0b3rt) [inline packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:11]: What is this TODO referring to?

File: packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx [hunk 21]
```diff
@@ -0,0 +1,230 @@
...
+          }
+        }
+        recommendationReason
+        yourVote
+        theirVote
+      }
+    }
+  `, {
+    variables: { userId: currentUser?._id, targetUserId: targetUser._id, limit:4 },
+  });
```
Comment (b0b3rt) [inline packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:170]: You probably want to skip this query if there is no current user.  (useQuery accepts a `skip` field, see `useDialogueMatchmaking` for examples)

File: packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx [hunk 22]
```diff
@@ -0,0 +1,230 @@
...
+
+  if (!currentUser) return <></>;
+  const preTopicRecommendations: {comment: {_id: string, contents: {html: string, plaintextMainText: string}}, recommendationReason: string, yourVote: string, theirVote: string}[] | undefined = topicData?.GetTwoUserTopicRecommendations; 
+  const topicRecommendations = preTopicRecommendations?.filter(topic => topic.theirVote !== null);
```
Comment (b0b3rt) [inline packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:174]: Are these going to be `null` or `undefined`?  Those don't get treated the same way in strict equality comparisons

File: packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx [hunk 23]
```diff
@@ -0,0 +1,230 @@
...
+  const numRecommendations = topicRecommendations?.length || 0;
```
Comment (b0b3rt) [inline packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:175]: nit: prefer `??`

File: packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx (hunk 16 above)
Comment (b0b3rt) [reply packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:205]: (also this entire section is pretty hard to read; would be good to extract various bits to top-level variable assignments instead of doing it all inline)

File: packages/lesswrong/components/dialogues/DialoguesList.tsx [hunk 24]
```diff
@@ -44,29 +46,18 @@ const styles = (theme: ThemeType) => ({
...
     color: theme.palette.lwTertiary.main,
     fontWeight: 645,
   },
-
   subheading: {
     marginTop: '10px',
   },
 
-  dialogueUserRow: {
+  dialogueUserRow: { // shared imports?
```
Comment (b0b3rt) [inline packages/lesswrong/components/dialogues/DialoguesList.tsx:60]: Does this require any follow-up?

File: packages/lesswrong/components/dialogues/DialoguesList.tsx [hunk 25]
```diff
@@ -180,71 +278,60 @@ const DialoguesList = ({ classes }: { classes: ClassesType<typeof styles> }) =>
...
+            <DialogueMatchRow key={index} rowProps={rowProps} classes={classes} showMatchNote={true} />
+          ))}
+          {currentUser?.showRecommendedPartners && recommendedDialoguePartnersRowPropsList?.map((rowProps, index) => (
+            <DialogueRecommendationRow key={index} rowProps={rowProps} showSuggestedTopics={showTopics} />
+          ))}
+        </div>}
       </AnalyticsContext>}
       
-      {dialoguePosts?.map((post, i: number) =>
+      {(!currentUser || currentUser?.showDialoguesList) && dialoguePosts?.map((post, i: number) =>
```
Comment (b0b3rt) [inline packages/lesswrong/components/dialogues/DialoguesList.tsx:327]: If you have `!currentUser || currentUser.showDialoguesList`, you don't need the null-safe access after the second `currentUser`, since the expression will short-circuit in the first half if `currentUser` doesn't exist, and Typescript is smart enough to figure that out

File: packages/lesswrong/components/dialogues/DialoguesList.tsx [hunk 26]
```diff
@@ -180,71 +278,60 @@ const DialoguesList = ({ classes }: { classes: ClassesType<typeof styles> }) =>
...
         <PostsItem
           key={post._id} post={post}
           showBottomBorder={i < dialoguePosts.length-1}
         />
       )}
 
-      {renderMyDialogues && (
+      {renderMyDialogues && currentUser?.showMyDialogues && (
```
Comment (b0b3rt) [inline packages/lesswrong/components/dialogues/DialoguesList.tsx:334]: You should just change `renderMyDialogues` to include `currentUser?.showMyDialogues` as part of the derived value

File: packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx (hunk 20 above)
Comment (lagerros) [reply packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:11]: Wondering if I should be sharing classes across files instead of having repeated code. Probably should

Comment (lagerros) [reply packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:11]: and I guess the way to do that is to pass a classes argument into the component

File: packages/lesswrong/components/dialogues/DialoguesList.tsx (hunk 25 above)
Comment (lagerros) [reply packages/lesswrong/components/dialogues/DialoguesList.tsx:327]: cool!

File: packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx (hunk 22 above)
Comment (lagerros) [reply packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:174]: null (just double checked)

File: packages/lesswrong/server/resolvers/userResolvers.ts (hunk 14 above)
Comment (lagerros) [reply packages/lesswrong/server/resolvers/userResolvers.ts:549]: That's cool to know! I actually find it marginally more legible and compatible with future ideas I migth have, to keep the randomisation inside of userResolvers, so will keep it there for now

File: packages/lesswrong/lib/components.ts (hunk 12 above)
Comment (lagerros) [reply packages/lesswrong/lib/components.ts:447]: No it did. I'm not quite sure what happened here and why it's being tracked as an addition.

File: packages/lesswrong/components/hooks/useDialogueMatchmaking.ts (hunk 8 above)
Comment (lagerros) [reply packages/lesswrong/components/hooks/useDialogueMatchmaking.ts:56]: we do want it to be dependent on that public db variable, though?

File: packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx (hunk 20 above)
Comment (lagerros) [reply packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:11]: actually, leaving this for now, since am making changes in the child file to the styles such that the added complexity of DRY doesn't seem worth it

File: packages/lesswrong/components/dialogues/DialoguesList.tsx (hunk 5 above)
Comment (lagerros) [reply packages/lesswrong/components/dialogues/DialoguesList.tsx:151]: request fast follow

doesn't seem blocking for shipping, and is finicky to unwrangle (it involves asking about some other types you wrote for the getRowProps functin, that I don't understand)

**Review by Discordius — APPROVED** [review]

File: packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx (hunk 22 above)
Comment (b0b3rt) [reply packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:187]: I forgot that we already have a helper function for this (`filterNonnull`).  Doesn't need to be fixed before merging but would put it on the list for a follow-up

File: packages/lesswrong/components/dialogues/DialoguesList.tsx (hunk 5 above)
Comment (b0b3rt) [reply packages/lesswrong/components/dialogues/DialoguesList.tsx:151]: sure, followup in another PR is fine

File: packages/lesswrong/components/hooks/useDialogueMatchmaking.ts [hunk 27]
```diff
@@ -43,6 +46,15 @@ export const useDialogueMatchmaking = <T extends MatchmakingProps>(props: T): Us
...
   `, { skip: skipByDefault || !getMatchedUsers });
 
+  const recommendedUsersQueryResult = useQuery(gql`
+  query GetDialogueMatchedUsers {
+    GetDialogueRecommendedUsers {
+      _id
+      displayName
+    }
+  }
+  `, { skip: skipByDefault || !getRecommendedUsers || !dialogueMatchmakingEnabled.get() });
```
Comment (b0b3rt) [inline packages/lesswrong/components/hooks/useDialogueMatchmaking.ts:56]: `dialogueMatchmakingEnabled.get()` is already covered in `skipByDefault`

File: packages/lesswrong/server/resolvers/userResolvers.ts (hunk 14 above)
Comment (b0b3rt) [reply packages/lesswrong/server/resolvers/userResolvers.ts]: @lagerros This kind of thing can be quite bad for performance so I would strongly recommend against unless you have an immediate use-case which requires pulling 100 users from the DB
